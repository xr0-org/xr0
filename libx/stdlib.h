#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
