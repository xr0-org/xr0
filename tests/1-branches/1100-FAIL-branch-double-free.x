#include <stdlib.h>

void
foo(int a, int b)
{
       int *p;
       p = malloc(1);
       if (!a) {
		if (!b) {
			free(p);
		}
       }
       if (a) {
               free(p);
       }
       if (b) {
               free(p);
       }
}
