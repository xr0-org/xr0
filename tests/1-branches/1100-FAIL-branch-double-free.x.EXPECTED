1-branches/1100-FAIL-branch-double-free.x:14:1:foo | $0 | $1: 
       stdlib.h:10:31:free: double free
